ls1intum/Artemis
# `Development`: Throw an exception if an email could not be sent to a user

### Checklist
#### General


#### Server

### Motivation and Context
Currently, there only was a `warn`-log when a mail could not be sent.
This caused issues as a wrong mail-config was not detected (the exception was catched).

### Description
This PR changes the log level to `error` and rethrows the exception as `ArtemisMailException` (a `RuntimeException`).

### Steps for Testing
On TS0/TS1:
1. Enable some email notifications (when an exercise is created in your settings)
2. Trigger that email notification (create a **text** exercise)
3. Make sure that the email is received

On TS3:
Repeat steps 1 & 2 and make sure that the error is logged with the `error`-level.

### Review Progress





#### Code Review
#### Manual Tests
